tests/run-make/static-pie/rmake.rs

```diff
@@ -0,0 +1,67 @@
+// How to manually run this
+// $ ./x.py test --target x86_64-unknown-linux-[musl,gnu] tests/run-make/static-pie
+
+//@ only-x86_64
+//@ only-linux
+//@ ignore-32bit
+
+use run_make_support::llvm_readobj;
+use run_make_support::rustc;
+use run_make_support::{cmd, target};
+
+// Minimum major versions supporting -static-pie
+const GCC_VERSION: u32 = 8;
+const CLANG_VERSION: u32 = 9;
+
+// Return `true` if the `compiler` version supports `-static-pie`.
+fn ok_compiler_version(compiler: &str) -> bool {
+    let version_threshold = match compiler {
+        "clang" => CLANG_VERSION,
+        "gcc" => GCC_VERSION,
+        other => panic!("unexpected compiler '{other}', expected 'clang' or 'gcc'"),
+    };
+
+    let process = cmd(compiler).arg("-dumpversion").run_unchecked();
+    if !process.status().success() {
+        eprintln!("No {compiler} version detected");
+        return false;
+    }
+    // 'major.minor.patch', 'major.minor', or 'major'
+    let version: u32 = process.stdout_utf8().split(".").next().unwrap().parse().unwrap();
```

Comment:
This uses the `-dumpversion` flag of `clang` and `gcc` instead of messing with the compiler preprocessor. It might not be supported in older versions?